mysql/cloudmysql/cloudmysql.go

```diff
@@ -79,6 +80,28 @@ func Open(ctx context.Context, certSource proxy.CertSource, params *Params) (*sq
 	return sql.OpenDB(connector(cfg.FormatDSN())), nil
 }
 
+// OpenGAE opens a Cloud SQL database on Google App Engine (GAE) using the
+// environment variables $DB_USER, $DB_DATABASE, $DB_INSTANCE, and
+// $DB_PASSWORD.
+func OpenGAE() (*sql.DB, error) {
```

Comment:
I'm similarly put off by the fact that the App Engine codepath would be different and that the application has to do App Engine specific things. Is there a compelling reason to do it this way?

Reply:
I guess there is no compelling reason. It would probably be better to factor out a function like `GetParamsFromEnv()` and have its output passed into `Open`. Trying that now.

Reply:
Okay, I changed it. I'm not thrilled about how the new code checks $GAE_ENV in three places total for the guestbook sample, but that issue will go away once the `go111` runtime supports command line flags.